web/html/src/manager/content-management/shared/components/panels/promote/promote.js

```diff
@@ -37,13 +38,18 @@ const Promote = (props: Props) => {
   }, [open]);
 
   const modalNameId = `${props.environmentPromote.label}-cm-promote-env-modal`;
+
   const disabled =
     !hasEditingPermissions
     || !props.environmentPromote.version
-    || props.environmentPromote.version <= props.environmentTarget.version;
+    || props.environmentPromote.version <= props.environmentTarget.version
+    || props.environmentPromote.status === "building"
+    || props.environmentTarget.status === "building"
+    || (props.environmentNextTarget || {}).status === "building";
+
   return (
     <div
-      {...(disabled ? {title: t("No version to promote")} : {})}
+      {...(disabled ? {title: t("No version to promote or colliding environment build in progress")} : {})}
```

Comment:
Good idea, I like the fact we give a hint in the UI, I'd also suggest to add a clear note in the documentation about the entire behavior. :+1: